stl/inc/atomic

```diff
@@ -161,7 +161,11 @@ _Ty kill_dependency(_Ty _Arg) noexcept { // "magic" template that kills dependen
 // FUNCTION _Check_memory_order
 inline void _Check_memory_order(const memory_order _Order) noexcept {
     // check that _Order is a valid memory_order
+#if _HAS_CXX20
+    if (_Order > memory_order_seq_cst) {

```

Comment:
Does this compile? We haven't defined `>` for `enum class memory_order` - and a scoped enumeration type doesn't implicitly convert to an integer type - so AFAICS it should not. Line 167, on the other hand, seems to do the right thing for both language modes by explicitly converting to `unsigned int`.

Reply:
This also just seems kinda needless -- the static_cast to unsigned int is correct in both modes. Worst case we might need to add a cast to the `memory_order_seq_cst` but that's equivalent in both modes.

Reply:
This does compile.  Line 167 does not compile with the `enum class` implementation of `memory_order`.

However, this implementation of the comparison would compile in all modes:

```if (static_cast<unsigned int>(_Order) > static_cast<unsigned int>(memory_order_seq_cst)) {```

Reply:
> This does compile.

Yes, sorry - I am 100% wrong. I strongly favor the "cast both sides in both language modes" solution to avoid conditional compilation here.

Reply:
Changed to cast both.